src/openfermion/circuits/low_rank.py

```diff
@@ -206,6 +206,18 @@ def prepare_one_body_squared_evolution(one_body_matrix, spin_basis=True):
                                                  numpy.eye(2))
         eigenvalues = numpy.kron(eigenvalues, numpy.ones(2))
 
+    # If the truncation reference is valid, 
+    # build truncated eigenvalues and eigenvectors
+    # based on the reference for density matrix
+    if truncation_ref is not None and isinstance(truncation_ref, float):
```

Comment:
Let's rely on python duck typing for the type of `truncation_ref` instead of mandating that it is a float. E.g.: what if it's a `np.float64`?